models/thales/ethereum/thales_ethereum_airdrop_claims.sql

```diff
@@ -0,0 +1,50 @@
+{{
+    config(
+        alias='airdrop_claims',
+        materialized = 'incremental',
+        file_format = 'delta',
+        tags=['static'],
+        unique_key = ['recipient', 'tx_hash', 'evt_index'],
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "thales",
+                                \'["hildobby"]\') }}'
+    )
+}}
+
+WITH early_price AS (
+    SELECT MIN(hour) AS hour
+    , MIN_BY(median_price, hour) AS price
+    FROM {{ ref('dex_prices') }}
+    WHERE blockchain = 'ethereum'
+    AND contract_address='0x03e173ad8d1581a4802d3b532ace27a62c5b81dc'
+    )
+
+, late_price AS (
+    SELECT MAX(hour) AS hour
+    , MAX_BY(median_price, hour) AS price
+    FROM {{ ref('dex_prices') }}
+    WHERE blockchain = 'ethereum'
+    AND contract_address='0x03e173ad8d1581a4802d3b532ace27a62c5b81dc'
+    )
+
+SELECT 'ethereum' AS blockchain
+, t.evt_block_time AS block_time
+, t.evt_block_number AS block_number
+, 'Thales' AS project
+, 'Thales Airdrop' AS airdrop_identifier
+, t.claimer AS recipient
+, t.contract_address
+, t.evt_tx_hash AS tx_hash
+, CAST(t.amount AS DECIMAL(38,0)) AS amount_raw
+, CAST(t.amount/POWER(10, 18) AS double) AS amount_original
+, CASE WHEN t.evt_block_time >= (SELECT hour FROM early_price) AND t.evt_block_time <= (SELECT hour FROM late_price) THEN CAST(pu.median_price*t.amount/POWER(10, 18) AS double)
+    WHEN t.evt_block_time < (SELECT hour FROM early_price) THEN CAST((SELECT price FROM early_price)*t.amount/POWER(10, 18) AS double)
+    WHEN t.evt_block_time > (SELECT hour FROM late_price) THEN CAST((SELECT price FROM late_price)*t.amount/POWER(10, 18) AS double)
+    END AS amount_usd
+, '0x03e173ad8d1581a4802d3b532ace27a62c5b81dc' AS token_address
+, 'THALES' AS token_symbol
+, t.evt_index
+FROM {{ source('thales_ethereum', 'Airdrop_evt_Claim') }} t
+LEFT JOIN {{ ref('dex_prices') }} pu ON pu.blockchain = 'ethereum'
```

Comment:
this join appears to be lacking join conditions, therefore performing joins which explode record count, i.e. why it's timing out